### ISSUE TITLE
Additional tests for 2D/3D REAL/LREAL arrays

### ISSUE THREAD

**Issue by sagatowski** [tcunit/TcUnit-Verifier#4]
As described in pull-request #3, there are some additional tests necessary to fully verify that the Assert-functionality for the 2D/3D REAL/LREAL arrays are fully working. Pull-request has been closed so the remaining tests are going to be handled through this issue.

I'll copy and paste the text from the PR here:

I've reviewed the code now and have some comments:
Some tests don't have complete arrays, many of them are just filled up with zeroes so it's just a zero to zero comparison which is not optimal.
See for example:
Test_LREAL_Array2d_DifferInSize
which has
a : ARRAY[-4..3, -1..2] OF LREAL := [64, 98, -32768, 32767, 5478, -378, 42, 6234];

This only fills upp a small amount of the 2D-array. Same thing goes for the 3D-arrays.

Could you update the tests with values where they are filled up?
I'll commit my changes and then I'll leave this PR open so you can do a new fork on the new changes and add these changes. Once this is done I'll close the PR.

I also think more tests are necessary for the 2D and 3D-arrays to proove that the method that asserts these take the x,y,z into consideration when checking whether the content is different. This declaratation and single test is not enough:

```
METHOD PRIVATE Test_LREAL_Array3d_DifferInContent
VAR
    a : ARRAY[-5..-4,-1..0,0..1] OF LREAL := [42, -23, -32768];
    b : ARRAY[1..2,4..5,6..7] OF LREAL := [42, 24, -32768];
END_VAR
```
First it doesn't even fill the whole array, second it's only a single test that doesn't take into consideration all the different variants of differences (considering the fact that the assert-method that does this check actually iterates through all three dimensions)

**Comment by DavidHopkinsFbr** [comment]
I didn't think it was necessary to fill the whole array because the assertion fails as soon as it encounters any different element value, and only reports the first difference it finds (which IIRC is consistent with the reporting behaviour of the other array assertions).

I see your point about needing to test the inner loops more exhaustively. I'll add test cases for detecting differences along the diagonal of each matrix, which will include the boundary points of [LBound, LBound] and [UBound, UBound]. Would that be ok?

**Comment by sagatowski** [comment]
That sounds like an excellent suggestion!

The previous merging was a little bit of a nightmare (had to do "squash" for the TcUnit-Verifier, in which all the commits-history were combined into one (but then the commiter changed to my name).

With that being said, would it be OK for you to create a completely new fork (or just refresh the one you have) before commiting, so I can just use your commit directly?

**Comment by DavidHopkinsFbr** [comment]
Shouldn't be a problem. I'll make a fresh fork and a fresh PR.

**Comment by sagatowski** [comment]
Hi David! Just wanted to check whether you have made any progress, and if possible, if you have an approxi. date of when this would be available?

**Comment by DavidHopkinsFbr** [comment]
Wow, time flies. Thanks for the nudge

**Comment by DavidHopkinsFbr** [comment]
I've renewed my fork and should be able to fill out those test cases tomorrow, hopefully

**Comment by sagatowski** [comment]
Sounds great David, very much looking forward to your code!

**Comment by DavidHopkinsFbr** [comment]
My recent fork of TcUnitVerifier won't compile, and not because of my changes. Looks like there's something missing from upstream?

![image](https://user-images.githubusercontent.com/49743077/63484048-effba980-c4d0-11e9-9d20-379d1f059786.png)

![image](https://user-images.githubusercontent.com/49743077/63484064-06096a00-c4d1-11e9-90cb-44fd1ff21d7b.png)

**Comment by sagatowski** [comment]
Hi David!
You need to clone and install the latest (working trunk) of TcUnit (1.0.0.0):
https://github.com/tcunit/TcUnit

**Comment by sagatowski** [comment]
Solved in commit 6e950fc.
